packages/salesforcedx-sobjects-faux-generator/src/describe/sObjectDescribe.ts

```diff
@@ -132,44 +133,56 @@ export class SObjectDescribe {
   }
 
   public async describeSObjectBatchRequest(
-    types: string[]
+    types: string[], cancellationToken: CancellationToken | undefined
   ): Promise<SObject[]> {
+    let batchResponse: BatchResponse;
     try {
       const batchRequest = this.buildBatchRequestBody(types);
-      const batchResponse = await this.runRequest(batchRequest);
-
-      const fetchedObjects: SObject[] = [];
-      if (batchResponse && batchResponse.results === undefined) {
-        return Promise.resolve(fetchedObjects);
-      }
-
-      batchResponse.results.forEach((sr, i) => {
-        if (sr.result instanceof Array) {
-          if (sr.result[0].errorCode && sr.result[0].message) {
-            console.log(`Error: ${sr.result[0].message} - ${types[i]}`);
-          }
-        }
-        fetchedObjects.push(sr.result);
-      });
-      return Promise.resolve(fetchedObjects);
+      batchResponse = await this.runRequest(batchRequest);
     } catch (error) {
       const errorMsg = error.hasOwnProperty('body')
         ? error.body
         : error.message;
       return Promise.reject(errorMsg);
     }
-  }
 
-  public async fetchObjects(types: string[]): Promise<SObject[]> {
-    const batchSize = MAX_BATCH_REQUEST_SIZE;
-    const requests = [];
-    for (let i = 0; i < types.length; i += batchSize) {
-      const batchTypes = types.slice(i, i + batchSize);
-      requests.push(this.describeSObjectBatchRequest(batchTypes));
+    const fetchedObjects: SObject[] = [];
```

Comment:
Check as part of each batch request and throw a custom error to handle cancellation